source/speech.py

```diff
@@ -915,9 +930,24 @@ def speakTextInfo(info,useCache=True,formatConfig=None,unit=None,reason=controlT
 			if  command.command=="controlStart":
 				# Control fields always start a new chunk, even if they have no field text.
 				inTextChunk=False
-				fieldText=info.getControlFieldSpeech(command.field,newControlFieldStack,"start_relative",formatConfig,extraDetail,reason=reason)
+				tempTextList=[]
+				if not inClickable and formatConfig['reportClickable']:
+					states=command.field.get('states')
+					if states and controlTypes.STATE_CLICKABLE in states:
+						# We have entered an outer most clickable or entered a new clickable after exiting a previous one 
+						# Announce it if there is nothing else interesting about the field, but not if the user turned it off. 
+						presCat=command.field.getPresentationCategory(newControlFieldStack[0:],formatConfig,reason)
+						if not presCat or presCat is command.field.PRESCAT_LAYOUT:
+							tempTextList.append(controlTypes.stateLabels[controlTypes.STATE_CLICKABLE])

```

Comment:
What if we keep a list of roles where all of that type of role is inherently clickable. That way we don't say something is a clibkable link or a clickable button? Or does this code already do that?

Reply:
We are already doing this by checking the field's presentationCategory.